### PR TITLE
MM-32547 - Remove any event from Timeline

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Can remove any event from the timeline
- Looks like:
image
image

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-32547

### CODE REVIEW

File: webapp/src/components/rhs/rhs_timeline_event_item.tsx [hunk 1]
```diff
@@ -75,6 +77,24 @@ const SummaryDetail = styled.div`
     color: var(--center-channel-color-64)
 `;
 
+const HoverMenu = styled.div`
```
Comment (crspeller) [inline webapp/src/components/rhs/rhs_timeline_event_item.tsx:80]: Should we share this? So when we get a new design we don't have to change it in multiple places?

**Review by crspeller — APPROVED** [review]
Nice! One non-blocking

File: webapp/src/components/rhs/rhs_timeline_event_item.tsx (hunk 1 above)
Comment (cpoile) [reply webapp/src/components/rhs/rhs_timeline_event_item.tsx:80]: I thought about it, but I wasn't sure if we wanted to make changes specific to just this (if I'm asked to from UI)... But, yah, let's worry about that if/when it happens. Will moved it to shared.

Comment (cpoile) [reply webapp/src/components/rhs/rhs_timeline_event_item.tsx]: Done, thanks!

File: server/api/incidents.go [hunk 2]
```diff
@@ -714,6 +715,21 @@ func (h *IncidentHandler) reminderButtonDismiss(w http.ResponseWriter, r *http.R
 	ReturnJSON(w, nil, http.StatusOK)
 }
 
+// removeTimelineEvent handles the DELETE /incidents/{id}/timeline/{eventID} endpoint.
+// User has been authenticated to edit the incident.
+func (h *IncidentHandler) removeTimelineEvent(w http.ResponseWriter, r *http.Request) {
```
Comment (agarciamontoro) [inline server/api/incidents.go:720]: We should probably update the api.yaml with this one

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Nice! :tada: The only thing I'm missing is the update of the API docs.

File: server/api/incidents.go (hunk 2 above)
Comment (cpoile) [reply server/api/incidents.go:720]: Good call, added. Thanks!

**Review by agarciamontoro — APPROVED** [review]
Great, thanks!

**Review by prapti — APPROVED** [review]
Tested and verified that removing the incident from timeline works as expected. LGTM!
Added the following tests:
1. Hovering over timeline event shows a delete hover button
2. Clicking on delete hover button shows a “Confirm Entry Delete” confirmation dialog
3. Clicking “Cancel” in “Confirm Entry Delete” confirmation dialog does not delete the timeline event
4. Clicking “Delete Entry” in “Confirm Entry Delete” confirmation dialog deletes the timeline event
5. Timeline event can be deleted by any incident channel member
6. A guest user can delete a timeline event (this will need to be tested later)

**Pull request merged**